plugins/logs/service_events

```diff
@@ -330,10 +368,17 @@ function fetch() {
         logmatch="${LOGFILEMAP[$n]}_regex"
         matches="$(tail -n +"$prvlines" "$logfile" | grep -Ec "${!logmatch}" || true)"
 
-        # Aggregate and add to the correct service counter
-        svc_counter="${svcnm}_total"
-        !((matches+=${!svc_counter}))
-        typeset "$svc_counter=$matches"
+        # If there were matches, aggregate them and add this log to the extinfo for the service
+        if [ "$matches" -gt 0 ]; then
+            # Aggregate and add to the correct service counter
+            svc_counter="${svcnm}_total"
+            !((matches+=${!svc_counter}))
```

Comment:
What is the purpose of the first exclamation mark here?
Maybe it would increase readability, if you would split this into two lines?
```shell
svc_counter_value=${!svc_counter}
matcher=$((matcher + svc_counter_value))
```

Reply:
Probably a good idea. The exclamation point is a weird hack that's necessary when the result of the expression equals 1. For whatever reason, in that case (and only that case), the expression returns an exit value of `1`, which, when `set -e` is used, kills the script without explanation. Negating the expression causes bash not to kill script execution, whatever the return value of the expression. Oh, bash....

But you're right. I'll break this out so it doesn't rely on idiomatic hacks.